lux-power-distribution-card.js

```diff
@@ -3,99 +3,73 @@ import * as hf from "./html-functions.js";
 import * as constants from "./constants.js";
 
 class LuxPowerDistributionCard extends HTMLElement {
+  constructor() {
+    super();
+    this.attachShadow({ mode: 'open' });
+  }
+
   set hass(hass) {
     this._hass = hass;
 
-    if (!this.card) {
+    if (!this.content) {
       this.createCard();
-      this.bindRefresh(this.card, this._hass, this.config);
-      this.bindHistoryGraph(this.card, this.config);
+      this.bindRefresh(this.card, this._hass, this._config);
+      this.bindHistoryGraph(this.card, this._config);
     }
 
     this.updateCard();
   }
 
   setConfig(config) {
-    if (!this.old_config) {
-      this.old_config = cef.buildConfig(config);
-    }
-    this.config = cef.buildConfig(config);
-    if (this.old_config != this.config) {
-      this.old_config = this.config;
-      this.createCard();
-    }
-
-    // console.log(this.config);
+    this._config = cef.buildConfig(config);
   }
 
   createCard() {
-    if (this.card) {
-      this.card.remove();
-    }
-
-    this.card = document.createElement("ha-card");
-    if (this.config.header) {
-      this.card.header = this.config.header;
+    if (!this._hass || !this._config) {
+      return;
     }
 
-    const content = document.createElement("div");
-    this.card.appendChild(content);
-
-    this.styles = document.createElement("style");
-    this.card.appendChild(this.styles);
-
-    this.appendChild(this.card);
+    const shadowRoot = this.shadowRoot;
 
-    // Set card base
-    content.innerHTML = hf.card_base;
-
-    // Generate Styles
-    this.styles.innerHTML = hf.generateStyles(this.config);
-
-    // Generate Status
-    const inv_info_element = this.card.querySelector("#taskbar-grid");
-    if (inv_info_element) {
-      inv_info_element.innerHTML = hf.generateStatus(this.config);
+    const card = document.createElement('ha-card');

```

Comment:
Apologies for the delayed feedback. I implemented this branch and I get an error here. The solutions was to change 
`const card = ` to `this.card = `
I'm not sure which is better for the card, to use a const at first and then assign, or directly initiate the card as part of the class. I'm reading up on the functions and implementations that you have that I don't know, once I'm done there and this part is fixed, I'll gladly merge this PR and then focus on the other one

Reply:
No problem, sorry as per my other comment, I didn't commit the change to `this.card` correctly.
It needs to be an instance attribute on the class so It can passed to the history graph functions.
We could return the const card, out of the createCard function, but it's completely fine to use instance attributes.